### ISSUE TITLE
sp_BlitzWho reports that a query is running as 'RCSI' when running a transaction at the default isolation level with only ALLOW_SNAPSHOT_ISOLATION turned ON.

### ISSUE THREAD

**Issue by Mitch-Wheat** [BrentOzarULTD/SQL-Server-First-Responder-Kit#2223]
**Version of the script**
@Version = '7.91', @VersionDate = '20191202'

**What is the current behavior?**
Not sure if this is a bug, or my misunderstanding: Creating a new database, and only turning on ALLOW_SNAPSHOT_ISOLATION, sp_BlitzWho reports that the query is running as 'Read Committed Snapshot Isolation' when running a transaction at the default isolation level.

**If the current behavior is a bug, please provide the steps to reproduce.**

    set nocount on
    GO

    create database TestRCSI
    GO

    select name, snapshot_isolation_state, snapshot_isolation_state_desc, is_read_committed_snapshot_on from sys.databases 
    where name = 'TestRCSI'
    GO

    use TestRCSI
    GO

    create table T1
    (
        id int not null identity(1,1) primary key,
        msg varchar(200) null,
        colId int null
    );
    GO

    -- fill table with some data...
    INSERT INTO T1 (msg, colId) VALUES (CONVERT(varchar(255), NEWID()), 10)
    GO 1000

    ------------------------

    -- This uses RC, as expected...
    BEGIN TRAN Test1

            update T1 set ColId=999 where ColId=10

            waitfor delay '00:00:15'

            -- run in another session: select * from T1 where colId = 10
            -- run in another session: exec dbo.sp_BlitzWho  @expertmode = 1
        
    ROLLBACK TRAN

    ------------------------

    -- Alter database to enable ONLY snapshot isolation...
    ALTER DATABASE TestRCSI  
    SET ALLOW_SNAPSHOT_ISOLATION ON  
    GO

    select name, snapshot_isolation_state, snapshot_isolation_state_desc, is_read_committed_snapshot_on from sys.databases 
    where name = 'TestRCSI'
    GO

    ------------------------
    -- This uses RCSI according to sp_BlitzWho...?? This was not the expexcted behaviour
    BEGIN TRAN Test2

            update T1 set ColId=999 where ColId=10

            waitfor delay '00:00:15'

            -- run in another session: select * from T1 where colId = 10
            -- run in another session: exec dbo.sp_BlitzWho  @expertmode = 1
        
    ROLLBACK TRAN
    ---------------------------------------------
    -- e.g.
    SELECT
        CASE transaction_isolation_level 
            WHEN 1 THEN 'ReadUncommitted' 
            WHEN 2 THEN 'ReadCommitted' 
            WHEN 3 THEN 'RepeatableRead' 
            WHEN 4 THEN 'Serializable' 
            WHEN 5 THEN 'Snapshot' 
            ELSE 'Unspecified'
        END as transaction_isolation_level_desc,
        *
    FROM sys.dm_exec_sessions
    WHERE security_id != 0x01
    GO

    -- e.g.
    exec master.dbo.sp_blitzWho @expertmode = 1
    GO

**What is the expected behavior?**
I would have thought that Test2's transaction isolation level be reported as 'Read Committed'?

**Which versions of SQL Server and which OS are affected by this issue? Did this work in previous versions of our procedures?**
Ran against Microsoft SQL Server 2016 (SP2-GDR) (KB4505220) - 13.0.5101.9 (X64)

**Comment by BrentOzar** [comment]
Hi Mitch. To see what data we're using to analyze the isolation levels, have you popped open the T-SQL and looked at the contents? sp_BlitzWho is pretty straightforward, and I don't want to hold you back - you can probably get the answers you're looking for more quickly that way, especially during the holiday season.

In this case, here's the relevant line of the code:

`WHEN s.transaction_isolation_level = 2 AND EXISTS (SELECT 1 FROM sys.dm_tran_active_snapshot_database_transactions AS trn WHERE s.session_id = trn.session_id AND is_snapshot = 0 ) THEN ''Read Committed Snapshot Isolation''`

So basically, we're calling it RCSI because SQL Server is generating version store rows for that transaction. Your transaction didn't specifically ASK for snapshot isolation, so it doesn't seem fair to call it SI, but it IS generating versions, so we call it RCSI.

If you want it to be called something different, you can totally propose that and we can have a discussion around it.

**Comment by Mitch-Wheat** [comment]
HI Brent: that's a good point, I should have taken a look at the TSQL. I got caught up with other things, apologies.  I was surprised at the behaviour. I thought that only turning on 'ALLOW_SNAPSHOT_ISOLATION' (i.e. not RCSI) and emitting a query without changing the transaction isolation level (i..e default of RC), would not create any row versions. Am I completely misunderstanding the way that setting works?

**Comment by BrentOzar** [comment]
Understood. The point of this repo isn’t really for technical questions about how SQL Server works - can you post that as a question on a Q&A site like https://DBA.stackexchange.com? Thanks! (Out on vacation with my family and can’t really do justice to this question at the moment.)

---
Tiny glass keyboard
Typos flowing like rivers
As winter snow thaws
